modules/planning/tasks/optimizers/piecewise_jerk_speed/piecewise_jerk_speed_nonlinear_optimizer.cc

```diff
@@ -160,27 +160,29 @@ Status PiecewiseJerkSpeedNonlinearOptimizer::Process(
 
   // TODO(Jinyun): evaluate the performance of piecewise linear warm start st
   // profile, will try piecewise polynomial st profile
-  std::vector<std::vector<double>> warm_start;
-  SpeedData reference_speed_data = *speed_data;
-  for (int i = 0; i < num_of_knots; ++i) {
-    SpeedPoint sp;
-    if (!reference_speed_data.EvaluateByTime(i * delta_t, &sp)) {
-      std::string msg("retriving dp st data for speed optimization failed!");
-      AERROR << msg;
-      speed_data->clear();
-      return Status(ErrorCode::PLANNING_ERROR, msg);
+  if (piecewise_jerk_nonlinear_speed_config.use_warm_start()) {
+    std::vector<std::vector<double>> warm_start;
+    SpeedData reference_speed_data = *speed_data;
+    for (int i = 0; i < num_of_knots; ++i) {
+      SpeedPoint sp;
+      if (!reference_speed_data.EvaluateByTime(i * delta_t, &sp)) {
+        std::string msg("retriving dp st data for speed optimization failed!");
+        AERROR << msg;
+        speed_data->clear();
+        return Status(ErrorCode::PLANNING_ERROR, msg);
+      }
+      warm_start.emplace_back(
+          std::initializer_list<double>{sp.s(), sp.v(), sp.a()});
```

Comment:
`{}` already indicates `std::initializer_list`, so you don't need to specify it.